satpy/readers/viirs_atms_sdr_base.py

```diff
@@ -100,11 +100,11 @@ def __init__(self, filename, filename_info, filetype_info, **kwargs):
         super().__init__(filename, filename_info, filetype_info, **kwargs)
 
     def _parse_datetime(self, datestr, timestr):
-        try:
-            datetime_str = datestr + timestr
-        except TypeError:
-            datetime_str = (str(datestr.data.compute().astype(str)) +
-                            str(timestr.data.compute().astype(str)))
+        if not isinstance(datestr, str):
+            datestr = str(datestr.data.compute().astype(str))
```

Comment:
just out of curiosity, what is the type of `datestr.data.compute().astype(str)` ?

Reply:
Numpy array with dtype `<U8`.